packages/jsii/lib/assembler.ts

```diff
@@ -669,6 +661,20 @@ export class Assembler implements Emitter {
     return !allUndefined ? result.docs : undefined;
   }
 
+  /**
+   * Check that all parameters the doc block refers to with a @param declaration actually exist
+   */
+  private _validateReferencedDocParams(method: spec.Method, methodSym: ts.Symbol) {
+    const params = getReferencedDocParams(methodSym);
+    const actualNames = new Set((method.parameters || []).map(p => p.name));
+    for (const param of params) {
+      if (!actualNames.has(param)) {
+        this._diagnostic(methodSym.valueDeclaration, ts.DiagnosticCategory.Error,
```

Comment:
Could this be a warning instead? I mean - is there any reason to actually fail on this?

Reply:
It could, but I'm wary. In my experience, warnings don't get fixed. But sure, I'll throw in a warning.

Reply:
We can turn this into an error in a later version if you like (using. warning first gives some notice period, which is great for CDK releases :D)

Reply:
And - feel free to record this in an issue so we don't forget!